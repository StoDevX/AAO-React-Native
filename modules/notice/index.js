// @flow

export {LoadingView} from './loading'
export {NoticeView} from './notice'
