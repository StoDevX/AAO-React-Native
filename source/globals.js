// @flow

export const GH_PAGES_URL = (relativePath: string) => {
	return `https://stodevx.github.io/AAO-React-Native/${relativePath}`
}

export const GH_NEW_ISSUE_URL =
	'https://github.com/StoDevX/AAO-React-Native/issues/new'
