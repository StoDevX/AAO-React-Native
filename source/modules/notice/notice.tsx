import * as React from 'react'
import {
	ActivityIndicator,
	StyleProp,
	StyleSheet,
	Text,
	TextStyle,
	View,
	ViewStyle,
} from 'react-native'
import * as c from '../colors'
import {Button} from '../button'
import {Heading} from '../markdown'

const styles = StyleSheet.create({
	container: {
		flex: 1,
		paddingHorizontal: 30,
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: c.systemBackground,
	},
	text: {
		textAlign: 'center',
		color: c.label,
	},
	spinner: {
		alignItems: 'center',
		justifyContent: 'center',
		padding: 8,
	},
})

interface Props {
	buttonDisabled?: boolean
	header?: string
	text?: string
	style?: StyleProp<ViewStyle>
	spinner?: boolean
	buttonText?: string
	onPress?: () => unknown
	textStyle?: StyleProp<TextStyle>
}

export function NoticeView(props: Props): React.JSX.Element {
	let {header, text, style, textStyle} = props
	let {buttonDisabled, buttonText, onPress} = props
	let {spinner} = props

	return (
		<View style={[styles.container, style]}>
			{spinner ? <ActivityIndicator style={styles.spinner} /> : null}

			{header ? (
				<Heading level={1} style={textStyle}>
					{header}
				</Heading>
			) : null}

			<Text selectable={true} style={[styles.text, textStyle]}>
				{text || 'Notice!'}
			</Text>

			{buttonText ? (
				<Button
					disabled={buttonDisabled}
					onPress={onPress}
					title={buttonText}
				/>
			) : null}
		</View>
	)
}
