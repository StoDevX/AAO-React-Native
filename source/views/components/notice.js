// @flow
import * as React from 'react'
import {ActivityIndicator, StyleSheet, Text, View} from 'react-native'
import * as c from './colors'
import {Button} from './button'
import {Heading} from './markdown/heading'
import {Viewport} from './viewport'

const styles = StyleSheet.create({
	container: {
		flex: 1,
		paddingHorizontal: 16,
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: c.white,
	},
	text: {
		textAlign: 'center',
	},
	spinner: {
		alignItems: 'center',
		justifyContent: 'center',
		padding: 8,
	},
})

type Props = {
	buttonDisabled?: boolean,
	header?: string,
	text?: string,
	style?: any,
	spinner?: boolean,
	buttonText?: string,
	onPress?: () => any,
	textStyle?: any,
}

export function NoticeView(props: Props) {
	const {header, text, style, textStyle} = props
	const {buttonDisabled, buttonText, onPress} = props
	const {spinner} = props

	return (
		<Viewport
			render={() => (
				<View style={[styles.container, style]}>
					{spinner ? <ActivityIndicator style={styles.spinner} /> : null}

					{header ? (
						<Heading level={1} style={textStyle}>
							{header}
						</Heading>
					) : null}

					<Text selectable={true} style={[styles.text, textStyle]}>
						{text || 'Notice!'}
					</Text>

					{buttonText ? (
						<Button
							disabled={buttonDisabled}
							onPress={onPress}
							title={buttonText}
						/>
					) : null}
				</View>
			)}
		/>
	)
}
