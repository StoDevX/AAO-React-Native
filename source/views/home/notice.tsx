import React from 'react'
import {View, Text, StyleSheet} from 'react-native'
import * as c from '@frogpond/colors'
import sample from 'lodash/sample'
import {CELL_MARGIN} from './button'
import {isDevMode} from '@frogpond/constants'

let messages = [
	'☃️ An Unofficial App Project ☃️',
	'For students, by students',
	'By students, for students',
	'An unofficial St. Olaf app',
	'For Oles, by Oles',
	'☃️',
	'🦁',
	'Made with ❤️ in Northfield, MN',
]

if (isDevMode()) {
	messages = [
		...messages,
		'made with  ⃟ in Ñ̸̞͖̘̱̰̥͇̗̂͌̇̎͊ͯ̎̓̎ͥ̋̐ͤͪͭ̚͘͢͢ø̸̛̞͊̎ͩ̍̉̑ͯͫͥ̚͟ͅ ̱̬̹̱̦®̵̬͖͙̻̩͓̖̠͉͈͍̈́̅͂͛̅̀͗ͤ̓́͡†̵̧͙̥̫̫͎̘̩̲̥̖̈̌͋̀ͨ̑̽̍̆̓̒̒̄̈́͒̓̕͜ ͍̩̫̼ͅ˙̶͕̰̗͓̯̫̲̮͕̪̝͎̩̬̺̔ͯ̌̈̽̌ͨ͊͊͐̀͆̽̐̓̃́̚͢͟ ̞̞̤ƒ͚͙̤ͭͪ͑̄͆͑ͯ̆͗̆ͨ̍̀͟͢ ̙͎̝͕͔̠͉̩̯͕͚̗̤ͅî̹̗̩̫̝̝͙̠̹̣̺̤̆ͭ̾̋ͬ̂ͫ̃̏ͥͬ́͜͠é̚ ̸͔͕̗̞̰́̅̅͒ ̪̩̞̰̫͓̞̱̫̞̭̯¬ͫ̾̆ ̍ͣ̎̀ͫͪͪ̋͌̂ ̪̘̯̝̤͌̆ͮ̕͜͜͡∂̢̛͕̻͖̈͌ͮ̂̾ͪͪ̑͋͂̂̂̂̈́̈́̓̌̍̌͜͞ ͙̫̤',
		'made with ∆ in Ñø®†˙ƒîé¬∂',
		'Made with 🤞 in ⬆️🌾',
		'⬆️🌾=🐄🏫♥️',
	]
}

export function UnofficialAppNotice(): JSX.Element {
	return (
		<View style={[styles.wrapper, styles.background]}>
			<Text style={styles.text}>{sample(messages)}</Text>
		</View>
	)
}

const styles = StyleSheet.create({
	wrapper: {
		justifyContent: 'center',
		marginBottom: CELL_MARGIN,
		marginHorizontal: CELL_MARGIN,
	},
	background: {
		backgroundColor: c.secondarySystemFill,
		borderRadius: 7,
	},
	text: {
		color: c.secondaryLabel,
		padding: 8,
		textAlign: 'center',
	},
})
