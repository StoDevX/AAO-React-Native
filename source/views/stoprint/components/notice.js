// @flow

import React from 'react'
import {
	Platform,
	Text,
	ScrollView,
	StyleSheet,
	RefreshControl,
} from 'react-native'
import Icon from 'react-native-vector-icons/Ionicons'
import {NoticeView} from '@frogpond/notice'
import {sto} from '../../../lib/colors'
import {Timer} from '@frogpond/timer'

type Props = {
	buttonText: string,
	description?: string,
	header: string,
	onPress: () => any,
	refresh: () => any,
	text: string,
}

export class StoPrintNoticeView extends React.PureComponent<Props> {
	render() {
		const {buttonText, description, header, onPress, text} = this.props

		return (
			<Timer
				interval={5000}
				invoke={this.props.refresh}
				moment={false}
				render={({refresh, loading}) => (
					<ScrollView
						contentContainerStyle={styles.content}
						refreshControl={
							<RefreshControl onRefresh={refresh} refreshing={loading} />
						}
						showsVerticalScrollIndicator={false}
						style={styles.container}
					>
						<Icon
							color={sto.black}
							name={Platform.OS === 'ios' ? 'ios-print' : 'md-print'}
							size={100}
						/>
						<NoticeView
							buttonText={buttonText}
							header={header}
							onPress={onPress}
							style={styles.notice}
							text={text}
						/>
						{description ? (
							<Text style={styles.description}>{description}</Text>
						) : null}
					</ScrollView>
				)}
			/>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		backgroundColor: sto.white,
	},
	content: {
		flex: 1,
		alignItems: 'center',
		justifyContent: 'center',
	},
	description: {
		marginHorizontal: 30,
		marginVertical: 20,
		textAlign: 'center',
	},
	notice: {
		flex: 0,
	},
})
