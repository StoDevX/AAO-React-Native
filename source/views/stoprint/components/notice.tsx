import * as React from 'react'
import {
	Platform,
	RefreshControl,
	ScrollView,
	StyleSheet,
	Text,
} from 'react-native'
import Icon from 'react-native-vector-icons/Ionicons'
import {NoticeView} from '@frogpond/notice'
import {sto} from '../../../lib/colors'

type Props = {
	buttonText: string
	description?: string
	header: string
	onPress: () => void
	onRefresh?: () => void
	refreshing?: boolean
	text: string
}

export const StoPrintNoticeView = (props: Props): React.ReactElement => {
	let {buttonText, description, header, onPress, text, onRefresh, refreshing} =
		props
	let iconName = Platform.OS === 'ios' ? 'ios-print' : 'md-print'

	return (
		<ScrollView
			contentContainerStyle={styles.content}
			refreshControl={
				onRefresh && refreshing != null ? (
					<RefreshControl onRefresh={onRefresh} refreshing={refreshing} />
				) : undefined
			}
			showsVerticalScrollIndicator={false}
			style={styles.container}
		>
			<Icon color={sto.black} name={iconName} size={100} />
			<NoticeView
				buttonText={buttonText}
				header={header}
				onPress={onPress}
				style={styles.notice}
				text={text}
			/>
			{description ? (
				<Text style={styles.description}>{description}</Text>
			) : null}
		</ScrollView>
	)
}

const styles = StyleSheet.create({
	container: {
		backgroundColor: sto.white,
	},
	content: {
		flex: 1,
		alignItems: 'center',
		justifyContent: 'center',
	},
	description: {
		marginHorizontal: 30,
		marginVertical: 20,
		textAlign: 'center',
	},
	notice: {
		flex: 0,
	},
})
