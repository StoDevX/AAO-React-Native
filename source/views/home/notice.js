// @flow

import * as React from 'react'
import * as glamorous from 'glamorous-native'
import * as c from '../components/colors'
import sample from 'lodash/sample'

const messages = [
	'☃️ An Unofficial App Project ☃️',
	'For students, by students',
	'By students, for students',
	'An unofficial St. Olaf app',
	'For Oles, by Oles',
	'☃️',
	'🦁',
]

export function UnofficialAppNotice() {
	return (
		<glamorous.View justifyContent="center" marginHorizontal={10}>
			<glamorous.View backgroundColor="rgba(0,0,0,0.05)" borderRadius={7}>
				<glamorous.Text color={c.black25Percent} padding={8} textAlign="center">
					{sample(messages)}
				</glamorous.Text>
			</glamorous.View>
		</glamorous.View>
	)
}
