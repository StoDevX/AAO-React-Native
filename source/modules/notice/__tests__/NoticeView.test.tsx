import React from 'react'
import {StyleSheet} from 'react-native'
import {describe, expect, it} from '@jest/globals'

import TestRenderer from 'react-test-renderer'
import {NoticeView} from '../notice'

describe('NoticeView', () => {
	describe('when given no text to display', () => {
		it('displays "Notice!" as its text', () => {
			const tree = TestRenderer.create(<NoticeView />).toJSON()
			expect(tree).toMatchSnapshot()
		})
	})

	describe('when given text to display', () => {
		it('displays the text', () => {
			const tree = TestRenderer.create(<NoticeView text="foo bar" />).toJSON()
			expect(tree).toMatchSnapshot()
		})
	})

	describe('when given view style overrides', () => {
		it('applies view style overrides', () => {
			const styleOverride = StyleSheet.create({
				view: {
					padding: 31,
				},
			})
			const tree = TestRenderer.create(
				<NoticeView style={styleOverride.view} text="foo bar" />,
			).toJSON()
			expect(tree).toMatchSnapshot()
		})
	})

	describe('when instructed to display a spinner', () => {
		it('displays a spinner', () => {
			const tree = TestRenderer.create(
				<NoticeView spinner={true} text="foo bar" />,
			).toJSON()
			expect(tree).toMatchSnapshot()
		})
	})

	describe('when header text is given', () => {
		it('displays the header text', () => {
			const tree = TestRenderer.create(
				<NoticeView header="blammo" text="foo bar" />,
			).toJSON()
			expect(tree).toMatchSnapshot()
		})
	})

	describe('when buttonText is given', () => {
		it('displays a button with the buttonText in its title', () => {
			const tree = TestRenderer.create(
				<NoticeView buttonText="button text" text="foo bar" />,
			).toJSON()
			expect(tree).toMatchSnapshot()
		})
	})
})
