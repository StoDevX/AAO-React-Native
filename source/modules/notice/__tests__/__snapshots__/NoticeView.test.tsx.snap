// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeView when buttonText is given displays a button with the buttonText in its title 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": {
          "semantic": [
            "systemBackground",
          ],
        },
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      },
      undefined,
    ]
  }
>
  <Text
    selectable={true}
    style={
      [
        {
          "color": {
            "semantic": [
              "label",
            ],
          },
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    foo bar
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignSelf": "center",
        "backgroundColor": {
          "semantic": [
            "tertiarySystemFill",
          ],
        },
        "borderRadius": 6,
        "marginVertical": 10,
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#007aff",
            "fontSize": 17,
            "fontWeight": "500",
            "textAlign": "center",
          },
          null,
          [
            {
              "backgroundColor": "transparent",
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
              "letterSpacing": -0.32,
              "lineHeight": 21,
            },
            {
              "color": {
                "semantic": [
                  "label",
                ],
              },
            },
            null,
          ],
          null,
        ]
      }
    >
      button text
    </Text>
  </View>
</View>
`;

exports[`NoticeView when given no text to display displays "Notice!" as its text 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": {
          "semantic": [
            "systemBackground",
          ],
        },
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      },
      undefined,
    ]
  }
>
  <Text
    selectable={true}
    style={
      [
        {
          "color": {
            "semantic": [
              "label",
            ],
          },
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Notice!
  </Text>
</View>
`;

exports[`NoticeView when given text to display displays the text 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": {
          "semantic": [
            "systemBackground",
          ],
        },
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      },
      undefined,
    ]
  }
>
  <Text
    selectable={true}
    style={
      [
        {
          "color": {
            "semantic": [
              "label",
            ],
          },
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    foo bar
  </Text>
</View>
`;

exports[`NoticeView when given view style overrides applies view style overrides 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": {
          "semantic": [
            "systemBackground",
          ],
        },
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      },
      {
        "padding": 31,
      },
    ]
  }
>
  <Text
    selectable={true}
    style={
      [
        {
          "color": {
            "semantic": [
              "label",
            ],
          },
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    foo bar
  </Text>
</View>
`;

exports[`NoticeView when header text is given displays the header text 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": {
          "semantic": [
            "systemBackground",
          ],
        },
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      },
      undefined,
    ]
  }
>
  <Text
    selectable={true}
    style={
      [
        {
          "marginBottom": 4,
          "marginTop": 8,
        },
        [
          {
            "backgroundColor": "transparent",
            "color": {
              "semantic": [
                "label",
              ],
            },
            "fontFamily": "System",
            "fontSize": 20,
            "fontWeight": "600",
            "letterSpacing": 0.75,
            "lineHeight": 25,
          },
          undefined,
        ],
      ]
    }
  >
    blammo
  </Text>
  <Text
    selectable={true}
    style={
      [
        {
          "color": {
            "semantic": [
              "label",
            ],
          },
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    foo bar
  </Text>
</View>
`;

exports[`NoticeView when instructed to display a spinner displays a spinner 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": {
          "semantic": [
            "systemBackground",
          ],
        },
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  />
  <Text
    selectable={true}
    style={
      [
        {
          "color": {
            "semantic": [
              "label",
            ],
          },
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    foo bar
  </Text>
</View>
`;
